client/blocks/login/index.jsx

```diff
@@ -137,13 +137,17 @@ class Login extends Component {
 		if ( this.props.onTwoFactorRequested ) {
 			this.props.onTwoFactorRequested( authType );
 		} else {
+			const currentURLQueryParameters = this.props?.redirectTo
+				? Object.fromEntries( new URL( this.props.redirectTo ).searchParams.entries() )
+				: '';
 			page(
 				login( {
 					isJetpack: this.props.isJetpack,
 					isGutenboarding: this.props.isGutenboarding,
 					// If no notification is sent, the user is using the authenticator for 2FA by default
 					twoFactorAuthType: authType,
 					locale: this.props.locale,
+					isPartnerSignup: isPartnerSignupQuery( currentURLQueryParameters ),
```

Comment:
I haven't dug into this so I might be misunderstanding something but can we just use the [`isPartnerSignup`](https://github.com/Automattic/wp-calypso/blob/f80ee93ee354571f0dee5fa9d88f98d0a8879384/client/blocks/login/index.jsx#L540) prop directly?

For example:
```suggestion
					isPartnerSignup: this.props.isPartnerSignup
```

Its possible this was added after your original PR or maybe I'm missing something?

Reply:
Maybe I'm missing something too because I don't see the 'isPartnerSignup` prop at this point but I see `redirectTo`, and there, we use the `isPartnerSignupQuery` function to check if the queryString has `is_partner_signup` or if `redirect_to` matches specific URLs (like woocommerce.com/partner-signup)

Reply:
You can add a `debugger;` to the top of that function and have a look at the props. Is there a scenario where it's set right?

![Screenshot_2022-07-12_16-23-07](https://user-images.githubusercontent.com/811776/178422679-466f083d-cc9e-4e1d-b67f-b0ad96043218.png)

Reply:
You are totally right, I tried and debugged as you said and we can use `isPartnerSignup: this.props.isPartnerSignup`

This let us remove from making all the URL checks before 😄